include/uio.h

```diff
@@ -21,6 +22,20 @@ typedef struct uio {
   vm_map_t *uio_vmspace; /* destination address space */
 } uio_t;
 
+/* Convenience function for initializing uio, where iov has one buffer */
+void prepare_single_uio(uio_t *uio, iovec_t *iov, uio_op_t op, vm_map_t *vm_map,
+                        size_t offset, void *buffer, size_t buflen);
```

Comment:
I reckon it's more natural to specify `offset` (of type `off_t` !) as the last argument.